src/GafferVDB/SphereLevelSet.cpp

```diff
@@ -0,0 +1,176 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2020, Don Boogert. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of Don Boogert nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "GafferVDB/SphereLevelSet.h"
+#include "GafferVDB/Interrupter.h"
+
+#include "Gaffer/StringPlug.h"
+
+#include "IECoreScene/MeshPrimitive.h"
+
+#include "IECoreVDB/VDBObject.h"
+
+#include "openvdb/tools/LevelSetSphere.h"
+
+using namespace Gaffer;
+using namespace GafferScene;
+using namespace GafferVDB;
+using namespace Imath;
+using namespace IECore;
+using namespace IECoreScene;
+using namespace IECoreVDB;
+
+IE_CORE_DEFINERUNTIMETYPED( SphereLevelSet );
+
+size_t SphereLevelSet::g_firstPlugIndex = 0;
+
+SphereLevelSet::SphereLevelSet( const std::string &name )
+: ObjectSource( name, "vdb" )
```

Comment:
Would "sphere" be a better name for the object rather than "vdb"? I think what the object represents is probably more useful than the object type, especially as we add more level-set-creating nodes